### DISCUSSION TITLE
faster imagemagick operations for appending 2 png and 1 gif side by side without as gif

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#4297] [Help] [answered]
I have this operation and I was wondering if this ImageMagick script can be faster significantly?

The intention is to append 2 png and 1 gif side by side 

Inputs

[![png 1][1]][1]
[![png 2][2]][2]
[![png 3][3]][3]

Expected result.

[![enter image description here][4]][4]

```
time (convert sample/1/1.png sample/1/1.gif &&
convert sample/1/2.png sample/1/2.gif &&
convert sample/1/1.gif -repage 900x169 -coalesce \
        null: \( sample/1/2.gif -coalesce \) -geometry +300+0 -layers Composite -coalesce \
        null: \( sample/1/3.gif -coalesce \) -geometry +600+0 -layers Composite  -coalesce \
        result.gif && mogrify -layers 'optimize' -fuzz 10% result.gif)
```

```text
real    0m1.346s
user    0m1.835s
sys     0m0.399s
```

  [1]: https://i.stack.imgur.com/ZJqEZ.png
  [2]: https://i.stack.imgur.com/XANS9.png
  [3]: https://i.stack.imgur.com/892mh.gif
  [4]: https://i.stack.imgur.com/19b5J.gif

**Comment by fmw42** [comment, marked as answer]
This is one way. Extend your animated gif to 3x its width. Append your two png files with the same width and height xc:none image. Then do the layers composite to the appended image onto the expanded animation.

```
convert \
\( sample3.gif -coalesce -background none -gravity east -extent 900x169 +gravity \) \
null: \
\( sample1.png sample2.png -size 300x169 xc:none +append \) \
-compose over -layers composite -layers optimize \
result.gif

```

On my Mac Mini OSX Sierra, I get 

0m 0.688s

![result](https://user-images.githubusercontent.com/13458218/135552334-58d20acb-b2e6-4915-819e-8685d1afea4c.gif)
